service/src/main/java/org/sunbird/actor/location/LocationActor.java

```diff
@@ -86,7 +87,7 @@ private void createLocation(Request request) {
             String id = ProjectUtil.generateUniqueId();
             locationRequest.setId(id);
             Location location = mapper.convertValue(locationRequest, Location.class);
-            Response response = locationDao.create(location, request.getRequestContext());
+            Response response = locationService.createLocation(location, request.getRequestContext());
```

Comment:
We should move all these Location creation logic to service layer

Reply:
Move validateUpsertLocnReq either in request validator class or in service layer

Reply:
Also move saveDataToES method to dao and call it via service layer

Reply:
Resolving this. Not required to move this as other service calls will result in request validation each time